docs/architecture/adr-050-improved-trusted-peering.md

```diff
@@ -0,0 +1,59 @@
+# ADR 50: Improved Trusted Peering
+
+## Changelog
+* 22-10-2019: Initial draft
+
+## Context
+
+When `max_num_inbound_peers` or `max_num_outbound_peers` of a node reached, the node cannot spare more slots to any peer 
+by inbound or outbound. Therefore, after certain period of disconnection, any important peering can be lost indefinitely 
+because all slots consumed by other peers, and the node stop trying to dial the peer anymore.
+
+This is happening because of two reasons, exponential backoff and absence of unconditional peering feature for trusted peers.
+
+
+## Decision
+
+We would like to suggest solving this problem by introducing two parameters in `config.toml`, `wildcard_peer_ids` and 
+`maximum_dial_period`. 
+
+1) `wildcard_peer_ids`
```

Comment:
What about `private_peers_ids`? If we force the node to always connect to private peers, would that be sufficient?

Reply:
Private and wildcard are having completely different definition. Private means not want to gossip, wildcard means allow connection anytime regardless of max connection.

Putting wildcard into private causes mix of two definitions.

Reply:
I'm not sure wildcard is the best name here, either, since it often implies needing to match some pattern. I like "unconditional"--a word you used yourself a few paragraphs earlier!

Reply:
unconditional it is! thanks for advice!